.github/workflows/copy-site-to-sourceforge.yml

```diff
@@ -6,17 +6,23 @@ name: "R: Copy Site to SourceForge"
 on:
   workflow_dispatch:
     inputs:
-      release-version:
+      version:
         description: 'Release Version without (-SNAPSHOT)'
         required: true
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        required: true
+
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
-  cancel-in-progress: true
+#concurrency:
```

Comment:
There is this small issue left with `concurrency`. Without removing it, we get:
https://github.com/stoyanK7/checkstyle/actions/runs/4177134639
![Screenshot from 2023-02-14 20-39-15](https://user-images.githubusercontent.com/23459549/218839948-b024659d-53c9-4304-9ccd-5dc8c0191fd4.png)

All jobs who have to run in parallel are cancelled except one which is left to run.

Reply:
I am currently trying to understand how this concurrency works and what we could do here.

Reply:
Looks like you need to adjust concurrency settings, set cancel in progress to false

Reply:
@stoyanK7 just use some grouping that would make each job have its own group name, check docs on this for job number or something

Reply:
@stoyanK7 looks like you could do ${{ job.container.id }} to get a unique group for each job

Reply:
I assume we want to use `job.container.id` only when the event trigger is `workflow_call`(when it is reused/called from another workflow) and for the rest of the time leave as it is right now, correct?

Reply:
@nrmancuso Using this variable on a job level concurrent does not work:
https://github.com/stoyanK7/checkstyle/actions/runs/4192845855
```
Invalid workflow file: .github/workflows/release.yml#L56
The workflow is not valid. .github/workflows/release.yml (Line: 56, Col: 14): Unrecognized named-value: 'job'. Located at position 1 within expression: job.container.id .github/workflows/release.yml (Line: 65, Col: 17): Unrecognized named-value: 'job'. Located at position 1 within expression: job.container.id

```
```yml
  new-milestone-and-issue-in-other-repo:
    uses: ./.github/workflows/new-milestone-and-issue-in-other-repo.yml
    needs: create-github-release
    with:
      version: ${{ inputs.version }}
    permissions:
      contents: write
      issues: write
    concurrency:
      group: ${{ job.container.id }}
```

Reply:
This is not allowed as well

```yml
concurrency:
  - group: ${{ github.run_id }}
    if: ${{ github.event_name == 'workflow_call' }}
  - group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
    if: ${{ github.event_name == 'workflow_dispatch' }}
    cancel-in-progress: true
```

Reply:
```yml
  new-milestone-and-issue-in-other-repo:
    ...
    concurrency:
      group: a1
      cancel-in-progress: true

  publish-releasenotes-outside:
    ...
    concurrency:
      group: a2
      cancel-in-progress: true
```
still cancels them https://github.com/stoyanK7/checkstyle/actions/runs/4193090241

Reply:
> @nrmancuso Using this variable on a job level concurrent does not work:

Yeah, it makes sense that this variable is not available until we start contianer.

> still cancels them https://github.com/stoyanK7/checkstyle/actions/runs/4193090241

I am not sure about this; it looks like one was cancelled, but the other two jobs were skipped.

Reply:
@stoyanK7 I have a theory, but we would need to investigate; I notice that most/all workflows have a space in the name. I wonder if this is causing an issue (perhaps workflow name is getting dropped/truncated/whatever). Otherwise, there should be no issue since we have a group naming scheme like: `${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}`, so example would be `R: Copy Site to SourceForge-12741`

So, each concurrency group would be unique to that workflow. But - if we drop everything after the first space, then we get `R:-12741`, and all workflows would be trying to run in this group.

Reply:
Please try https://github.com/checkstyle/checkstyle/pull/12741/commits/74d9ab6c73b9f815bb2b15df0b94f0f72367ccf7#r1108301551 again with `cancel-in-progress: false`, workflows should be queued in this case, but not cancelled.

Reply:
@stoyanK7 OK,  here is the problem: https://docs.github.com/en/actions/using-workflows/reusing-workflows#:~:text=the%20github%20context%20is%20always%20associated%20with%20the%20caller%20workflow.

> the github context is always associated with the caller workflow.

We need to use some other context/means for naming

Reply:
> Please try https://github.com/checkstyle/checkstyle/commit/74d9ab6c73b9f815bb2b15df0b94f0f72367ccf7#r1108301551 again with cancel-in-progress: false, workflows should be queued in this case, but not cancelled if this is a valid grouping.

![Screenshot from 2023-02-17 16-10-10](https://user-images.githubusercontent.com/23459549/219691995-c9d7e82d-3f8c-462f-8417-4c6429dc270c.png)

https://github.com/stoyanK7/checkstyle/actions/runs/4205004190

Reply:
> You could also pass the name of the workflow itself as a parameter to the child workflow.

You mean like as an input?
```yml
... release.yml
  new-milestone-and-issue-in-other-repo:
    uses: ./.github/workflows/new-milestone-and-issue-in-other-repo.yml
    with:
      version: ${{ inputs.version }}
      concurrency-group: new-milestone
```

and then
```yml
... new-milestone-and-issue-in-other-repo.yml
on:
  workflow_call:
    inputs:
      version:
        type: string
        required: true
    concurrency-group:
        type: string
        required: true
        default: ""
```

This is what I assume you mean.

Reply:
Here is the result. It works. They ran simultaneously

![Screenshot from 2023-02-17 16-24-41](https://user-images.githubusercontent.com/23459549/219695699-63341262-64bd-40f4-aeab-41f1e4cfb922.png)

https://github.com/stoyanK7/checkstyle/actions/runs/4205128732

Reply:
Done. Pushed changes.